multibody/kinematic/kinematic_evaluator.h

```diff
@@ -72,17 +76,35 @@ class KinematicEvaluator {
 
   const std::vector<int>& active_inds() const;
 
-  int num_full() const {
-    return length_;
-  }
+  int num_full() const { return length_; }
+
+  int num_active() const { return num_active_; }
+
+  const drake::multibody::MultibodyPlant<T>& plant() const { return plant_; }
+
+  const std::vector<int>& active_inds() { return active_inds_; };
+
+  bool is_active(int index) const;
+
+  /// Convert an index, relative to the full index set, to be an active index.
+  /// Returns -1 if the given index is not active
+  int full_index_to_active_index(int full_index) const;
 
-  int num_active() const {
-    return num_active_;
-  }
+  /// Create a friction cone constraint on the force variables (associated with
+  /// the full Jacobian). Subclasses which might be associated with frictional
+  /// contact should implement this method.
+  virtual std::shared_ptr<drake::solvers::Constraint>
+  CreateConicFrictionConstraint(double mu) const {
```

Comment:
Is there going to be another evaluator that has friction constraint? If not, should we move the friction constraint methods to WorldPointEvaluator?